aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -15,122 +15,103 @@
 namespace at {
 namespace native {
 
-using sum_type = void(Tensor &, const Tensor &, size_t, bool);
-sum_type *sumImpl = &DispatchStub<sum_type>::init<sumImplC, &sumImpl>;
-
-Tensor _sum_cpu(const Tensor &self) {
-  if (self.is_contiguous()) {
-    Tensor result = self.type().tensor({});
-    sumImpl(result, self, 0, true);
-    return result;
-  }
-  return self._sumall();
-}
-
-Tensor _sum_cuda(const Tensor &self_) { return self_._sumall(); }
-
-Tensor sum(const Tensor &self, int64_t dim_, bool keepdim) {
+static Tensor _dimreduce_setup(const Tensor &self, int64_t dim_,
+                               int64_t ident) {
   int64_t dim = maybe_wrap_dim(dim_, self.dim());
   Tensor result = self.type().tensor();
   if (self.numel() == 1 && self.ndimension() == 0) {
     result.resize_({});
     result.fill_(self);
-    return result;
+    return std::move(result);
```

Comment:
Never `return std::move(...)`. This prevents return value optimization and ends up in slower code (clang even warns about it)